VirtualReality/Widgets/qMRMLVirtualRealityView.cxx

```diff
@@ -456,9 +462,65 @@ void qMRMLVirtualRealityViewPrivate::doOpenVirtualReality()
       this->Camera->GetViewPlaneNormal(this->LastViewDirection);
       this->Camera->GetViewUp(this->LastViewUp);
       this->Camera->GetPosition(this->LastViewPosition);
+
+      int disabledModify = this->MRMLVirtualRealityViewNode->StartModify();
+      updateTransformNodeWithControllerPose(vtkEventDataDevice::LeftController);
+      updateTransformNodeWithControllerPose(vtkEventDataDevice::RightController);
+      this->MRMLVirtualRealityViewNode->EndModify(disabledModify);
+
       this->LastViewUpdateTime->StartTimer();
     }
+  }
+}
+
+// --------------------------------------------------------------------------
+void qMRMLVirtualRealityViewPrivate::updateTransformNodeWithControllerPose(vtkEventDataDevice device)
+{
+  vtkMRMLLinearTransformNode* node = this->MRMLVirtualRealityViewNode->GetControllerTransformNode(device);
 
+  if (node == NULL)
+  {
+    qCritical() << "Unable to retrieve linear transform node for device: " << (int)device;
+    return;
+  }
+
+  if (this->RenderWindow->GetTrackedDeviceIndexForDevice(device) == vr::k_unTrackedDeviceIndexInvalid)
+  {
+    node->SetAttribute("VirtualReality.ControllerActive", "0");
+    node->SetAttribute("VirtualReality.ControllerConnected", "0");
+    return;
+  }
+
+  vr::TrackedDevicePose_t& pose = this->RenderWindow->GetTrackedDevicePose(this->RenderWindow->GetTrackedDeviceIndexForDevice(device));
+  if (pose.eTrackingResult != vr::TrackingResult_Running_OK)
+  {
+    node->SetAttribute("VirtualReality.ControllerActive", "0");
+  }
+  else
+  {
+    node->SetAttribute("VirtualReality.ControllerActive", "1");
+  }
+
+  if (!pose.bDeviceIsConnected)
+  {
+    node->SetAttribute("VirtualReality.ControllerConnected", "0");
+  }
+  else
+  {
+    node->SetAttribute("VirtualReality.ControllerConnected", "1");
+  }
+
+  double pos[3];
+  double ppos[3];
+  double wxyz[4];
+  double wdir[3];
+  this->Interactor->ConvertPoseToWorldCoordinates(pose, pos, wxyz, ppos, wdir);
+  vtkSmartPointer<vtkTransform> transform = vtkSmartPointer<vtkTransform>::New();
```

Comment:
in these kind of simple cases you can use vtkNew<> instead of vtkSmartPointer

Reply:
does vtkNew allocate on the stack?